core/src/wallet/ripple/RippleLikeAccount.cpp

```diff
@@ -0,0 +1,414 @@
+                                             const std::shared_ptr<RippleLikeBlockchainExplorer> &explorer,
+                                             const std::shared_ptr<RippleLikeBlockchainObserver> &observer,
+                                             const std::shared_ptr<RippleLikeAccountSynchronizer> &synchronizer,
+                                             const std::shared_ptr<RippleLikeKeychain> &keychain) : AbstractAccount(
+                wallet, index) {
+            _explorer = explorer;
+            _observer = observer;
+            _synchronizer = synchronizer;
+            _keychain = keychain;
+            _accountAddress = keychain->getAddress()->toString();
+            _currentLedgerSequence = 0;
+        }
+
+
+        FuturePtr<RippleLikeBlockchainExplorerTransaction> RippleLikeAccount::getTransaction(const std::string &hash) {
+            auto self = std::dynamic_pointer_cast<RippleLikeAccount>(shared_from_this());
+            return async<std::shared_ptr<RippleLikeBlockchainExplorerTransaction>>(
+                    [=]() -> std::shared_ptr<RippleLikeBlockchainExplorerTransaction> {
+                        auto tx = std::make_shared<RippleLikeBlockchainExplorerTransaction>();
+                        soci::session sql(self->getWallet()->getDatabase()->getPool());
+                        if (!RippleLikeTransactionDatabaseHelper::getTransactionByHash(sql, hash, *tx)) {
+                            throw make_exception(api::ErrorCode::TRANSACTION_NOT_FOUND, "Transaction {} not found",
+                                                 hash);
+                        }
+                        return tx;
+                    });
+        }
+
+        void RippleLikeAccount::inflateOperation(Operation &out,
+                                                 const std::shared_ptr<const AbstractWallet> &wallet,
+                                                 const RippleLikeBlockchainExplorerTransaction &tx) {
+            out.accountUid = getAccountUid();
+            out.block = tx.block;
+            out.rippleTransaction = Option<RippleLikeBlockchainExplorerTransaction>(tx);
+            out.currencyName = getWallet()->getCurrency().name;
+            out.walletType = getWalletType();
+            out.walletUid = wallet->getWalletUid();
+            out.date = tx.receivedAt;
+            if (out.block.nonEmpty())
+                out.block.getValue().currencyName = wallet->getCurrency().name;
+            out.rippleTransaction.getValue().block = out.block;
+        }
+
+        int RippleLikeAccount::putTransaction(soci::session &sql,
+                                              const RippleLikeBlockchainExplorerTransaction &transaction) {
+            auto wallet = getWallet();
+            if (wallet == nullptr) {
+                throw Exception(api::ErrorCode::RUNTIME_ERROR, "Wallet reference is dead.");
+            }
+
+            if (transaction.block.nonEmpty())
```

Comment:
The previous `if` at line _104_ uses curly braces for a single statement; I think you should stick to this coding style and use curly braces here as well. On a more general note, I think omitting curly braces can mislead people from time to time so we should stick to using curly braces for block constructs with one-liners.